File: frontend/src/types/employeePerformance.ts
export interface PerformanceKpiEntry {
  id: number;
  kpiDefinitionId: number;
  kpiName: string;
  measurementUnit: string | null;
  targetValue: string | number | null;
  weight: number | null;
  periodStart: string;
  periodEnd: string;
  actualValue: string | number;
  score: number | null;
  notes: string | null;
  createdAt: string;
}

export interface MonthlyPerformanceHistoryItem {
  periodYear: number;
  periodMonth: number;
  label: string;
  scoreOutOf10: number;
  entriesCount: number;
  finalized: boolean;
  aiSummary: string | null;
}

export interface EmployeePerformance {
  employeeId: number;
  employeeName: string;
  employeeCode: string;
  periodYear: number;
  periodMonth: number;
  periodLabel: string;
  isCurrentMonth: boolean;
  isFinalized: boolean;
  scoreOutOf10: number;
  overallPct: number | null;
  entriesCount: number;
  entries: PerformanceKpiEntry[];
  history: MonthlyPerformanceHistoryItem[];
  aiSummary: string | null;
}

export interface EmployeePerformanceAiSummary {
  employeeId: number;
  periodYear: number;
  periodMonth: number;
  scoreOutOf10: number;
  aiSummary: string;
}

File: backend/app/schemas/employee_performance.py
"""Employee Development — monthly performance score schemas."""
from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class PerformanceKpiEntryRead(BaseModel):
    id: int
    kpi_definition_id: int
    kpi_name: str
    measurement_unit: str | None = None
    target_value: Decimal | None = None
    weight: float | None = None
    period_start: date
    period_end: date
    actual_value: Decimal
    score: float | None = None
    notes: str | None = None
    created_at: datetime


class MonthlyPerformanceHistoryItem(BaseModel):
    period_year: int
    period_month: int
    label: str  # e.g. "May 2026"
    score_out_of_10: float
    entries_count: int
    finalized: bool = True
    ai_summary: str | None = None


class EmployeePerformanceRead(BaseModel):
    employee_id: int
    employee_name: str
    employee_code: str
    period_year: int
    period_month: int
    period_label: str
    is_current_month: bool
    is_finalized: bool
    score_out_of_10: float
    overall_pct: float | None = None
    entries_count: int
    entries: list[PerformanceKpiEntryRead] = []
    history: list[MonthlyPerformanceHistoryItem] = []
    ai_summary: str | None = None


class EmployeePerformanceAiSummaryRequest(BaseModel):
    employee_id: int
    period_year: int = Field(ge=2000, le=2100)
    period_month: int = Field(ge=1, le=12)


class EmployeePerformanceAiSummaryResponse(BaseModel):
    employee_id: int
    period_year: int
    period_month: int
    score_out_of_10: float
    ai_summary: str
